src/goto-symex/symex_atomic_section.cpp

```diff
@@ -57,11 +57,9 @@ void goto_symext::symex_atomic_end(statet &state)
         it != pair.second.second.end();
         ++it)
       read_guard|=*it;
-    exprt read_guard_expr=read_guard.as_expr();
-    do_simplify(read_guard_expr);
```

Comment:
Is it a functional change that we no longer simplify `read_guard` here?

Reply:
I suspect we ought to have a simplify(guardt &) method, which simplifies when `guardt` == `exprt` but is a no-op on a BDD, which is always inherently in simplified form